cockatrice/src/zoneviewwidget.cpp

```diff
@@ -185,6 +185,24 @@ void ZoneViewWidget::paint(QPainter *painter, const QStyleOptionGraphicsItem *op
 
 void ZoneViewWidget::moveWidget(QPointF scenePos)
 {
+    if(scenePos.x() < 0)
+    {
+        scenePos.setX(0);
+    } else {
+        qreal maxw = scene()->sceneRect().width() - 100;
+        if(scenePos.x() > maxw)
+            scenePos.setX(maxw);
+    }
```

Comment:
That's fair, I wasn't considering the frequency that it's triggered.

Keep it as it. And if you notice me adding code like that in often-called places please point it out during review..